pkg/autoscaler/autoscaler/pdplan_autoscaler.go

```diff
@@ -80,11 +82,51 @@ func (am *autoScalerManager) deleteAutoscalingClusters(tc *v1alpha1.TidbCluster,
 }
 
 func (am *autoScalerManager) updateAutoscalingClusters(tac *v1alpha1.TidbClusterAutoScaler, groups []string, groupTcMap map[string]*v1alpha1.TidbCluster, groupPlanMap map[string]pdapi.Plan) error {
-	// TODO in next PR
-	if !checkAutoScalingPrerequisites(nil, nil, v1alpha1.TiDBMemberType) {
-		return nil
+	var errs []error
+	for _, group := range groups {
+		actual, oldTc, expected := groupTcMap[group].DeepCopy(), groupTcMap[group], groupPlanMap[group]
+		component := expected.Component
+
+		if !checkAutoscalingComponent(tac, component) {
+			continue
+		}
+
+		switch component {
+		case v1alpha1.TiKVMemberType.String():
+			sts, err := am.stsLister.StatefulSets(actual.Namespace).Get(operatorUtils.GetStatefulSetName(actual, v1alpha1.TiKVMemberType))
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			if !checkAutoScalingPrerequisites(actual, sts, v1alpha1.TiKVMemberType) {
+				continue
+			}
+			if actual.Spec.TiKV.Replicas == int32(expected.Count) {
```

Comment:
So for the same group, the response from PD only differs in the count, right?

Reply:
Yes